### PR TITLE
Add section on getting absolute URLs via BigQuery dataset

### PR DESCRIPTION
**Pull request by di**

### CODE REVIEW

**Comment by dstufft** [comment]
I doubt we'd ever change it, but just to note that technically the URL that files are stored at are not guaranteed to be stable.

File: source/guides/analyzing-pypi-package-downloads.rst [hunk 1]
```diff
@@ -217,6 +217,40 @@ query processes over 500 GB of data.
...
 
+
+Getting absolute links to artifacts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It's sometimes helpful to be able to get the absolute links to download
+artifacts from PyPI based on their hashes, e.g. if a particular project or
+release has been deleted from PyPI. The metadata table includes the ``path``
+column, which includes the hash and artifact filename.
+
```
Comment (di) [inline source/guides/analyzing-pypi-package-downloads.rst:228]:
```suggestion

Note that the URL generated here is not guaranteed to be stable, but currently aligns with the URL where PyPI artifacts are hosted.
```

@dstufft This sufficient?

**Review by Harlee1994 — APPROVED** [review]

File: source/guides/analyzing-pypi-package-downloads.rst (hunk 1 above)
Comment (webknjaz) [reply source/guides/analyzing-pypi-package-downloads.rst:228]: FWIW, many downstream packaging ecosystems have macros/helpers for retrieving sdists from the PyPI's CDN. I'm pretty sure that they just do URL conversions to the jump/redirect URLs w/o making any queries...

File: source/guides/analyzing-pypi-package-downloads.rst [hunk 2]
```diff
@@ -217,6 +217,41 @@ query processes over 500 GB of data.
...
+Note that the URL generated here is not guaranteed to be stable, but currently aligns with the URL where PyPI artifacts are hosted.
```
Comment (webknjaz) [inline source/guides/analyzing-pypi-package-downloads.rst:229]: How about using a note-block for this?
```suggestion
.. note::
   The URL generated here is not guaranteed to be stable, but currently aligns with the URL where PyPI artifacts are hosted.
```

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@di approved and feel free to mark the comments as resolved if you're not planning to act on them.

File: source/guides/analyzing-pypi-package-downloads.rst [hunk 3]
```diff
@@ -217,6 +217,42 @@ query processes over 500 GB of data.
...
+Getting absolute links to artifacts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It's sometimes helpful to be able to get the absolute links to download
+artifacts from PyPI based on their hashes, e.g. if a particular project or
+release has been deleted from PyPI. The metadata table includes the ``path``
+column, which includes the hash and artifact filename.
+
+.. note::
+   The URL generated here is not guaranteed to be stable, but currently aligns with the URL where PyPI artifacts are hosted.
```
Comment (di) [inline source/guides/analyzing-pypi-package-downloads.rst:230]:
```suggestion
   The URL generated here is not guaranteed to be stable, but currently aligns with the URL where PyPI artifacts are hosted.

```

**Pull request merged**